geoutils/pointcloud/pointcloud.py

```diff
@@ -1 +1,393 @@
+    """Load point cloud data from LAS/LAZ/COPC file."""
+
+    # Read file
+    las = laspy.read(filename)
+
+    # Get data from requested columns
+    data = np.vstack([las[n] for n in columns]).T
+
+    # Build geodataframe
+    gdf = gpd.GeoDataFrame(
+        geometry=gpd.points_from_xy(x=las.x, y=las.y, crs=las.header.parse_crs(prefer_wkt=False)),
+        data=data,
+        columns=columns,
+    )
+
+    return gdf
+
+
+def _load_laspy_metadata(
+    filename: str,
+) -> tuple[CRS, int, BoundingBox, pd.Index]:
+    """Load point cloud metadata from LAS/LAZ/COPC file."""
+
+    with laspy.open(filename) as f:
+
+        crs = f.header.parse_crs(prefer_wkt=False)
+        nb_points = f.header.point_count
+        bounds = BoundingBox(left=f.header.x_min, right=f.header.x_max, bottom=f.header.y_min, top=f.header.y_max)
+        columns_names = pd.Index(list(f.header.point_format.dimension_names))
+
+    return crs, nb_points, bounds, columns_names
+
+
+# def _write_laspy(filename: str, pc: gpd.GeoDataFrame):
+#     """Write a point cloud dataset as LAS/LAZ/COPC."""
+#
+#     with laspy.open(filename) as f:
+#         new_hdr = laspy.LasHeader(version="1.4", point_format=6)
+#         # You can set the scales and offsets to values that suits your data
+#         new_hdr.scales = np.array([1.0, 0.5, 0.1])
+#         new_las = laspy.LasData(header = new_hdr, points=)
+#
+#     return
+
+
+class PointCloud(gu.Vector):  # type: ignore[misc]
+    """
+    The georeferenced point cloud.
+
+    A point cloud is a vector of 2D point geometries associated to numeric values from a data column, and potentially
+    auxiliary data columns.
+
+     Main attributes:
+        ds: :class:`geopandas.GeoDataFrame`
+            Geodataframe of the point cloud.
+        data_column: str
+            Name of point cloud data column.
+        crs: :class:`pyproj.crs.CRS`
+            Coordinate reference system of the point cloud.
+        bounds: :class:`rio.coords.BoundingBox`
+            Coordinate bounds of the point cloud.
+
+
+    All other attributes are derivatives of those attributes, or read from the file on disk.
+    See the API for more details.
+    """
+
+    def __init__(
+        self,
+        filename_or_dataset: str | pathlib.Path | gpd.GeoDataFrame | gpd.GeoSeries | BaseGeometry,
+        data_column: str,
+    ):
+        """
+        Instantiate a point cloud from either a data column name and a vector (filename, GeoPandas dataframe or series,
+        or a Shapely geometry), or only with a point cloud file type.
+
+        :param filename_or_dataset: Path to vector file, or GeoPandas dataframe or series, or Shapely geometry.
+        :param data_column: Name of main data column defining the point cloud.
+        """
+
+        self._ds: gpd.GeoDataFrame | None = None
+        self._name: str | None = None
+        self._crs: CRS | None = None
+        self._bounds: BoundingBox
+        self._data_column: str
```

Comment:
Just a general note. I don't find the names "data_column" and "all_columns" very intuitive. How did you pick the names? Hence I am not sure to understand what is which, even with the PR description. This will most likely be clearer after reading the code, but it should be fairly intuitive to the user without searching through doc or code.  

I don't know what is the most used term for this.
In OGR this was named "fields" and similarly in QGIS. So in the old geoutils, we stuck to this term. 
Would it also not be easier to save a "fields" attribute as a panda dataframe, so the column names and data can be retrieved from one object instead of two?

Reply:
Ok, looking at the code below it is a bit more clear the meaning of the "columns" attributes. I understand that it originates from Pandas columns. 
I also realize that in the Vector class (and so in PointCloud too), the field attributes are not directly accessible from the Vector object, but from `Vector.ds` only. There is also no equivalent to the "all_columns" attribute in Vector.
Sorry I don't have a perfect solution to give, but maybe we should discuss this more and make sure everything is clear and consistent.

Reply:
I've removed "all_columns" and simply override "columns" to stay consistent with Pandas/GeoPandas.
For "data_column": I don't like "z" too much because it's usually used for 3D points (elevation), while a "data" associated to 2D points is more generic.
We could use "z" for the elevation point cloud class?